drivers/motor/foc/foc_dev.c

```diff
@@ -147,19 +142,13 @@ static int foc_open(FAR struct file *filep)
 
           if (tmp == 1)
             {
-              /* Yes.. perform one time driver setup */
-
-              flags = enter_critical_section();
-
```

Comment:
@xiaoxiang781216 @Donny9 why the critical section is not necessary anymore? Should this be done by the lower half of the driver?

Reply:
The code is already locked by nxsem_wait.

Reply:
yes, I also think  the critical section should be done by lower half of the driver, more detailed control.

Reply:
Sure, let's fix it one by one.